src/pep8ext_naming.py

```diff
@@ -233,14 +228,14 @@ class ClassNameCheck(BaseASTCheck):
 
     Classes for internal use have a leading underscore in addition.
     """
-    check = MIXEDCASE_REGEX.match
     N801 = "class name '{name}' should use CapWords convention"
 
     def visit_classdef(self, node, parents, ignore=None):
-        if ignore and node.name in ignore:
+        name = node.name
+        if ignore and name in ignore:
             return
-        if not self.check(node.name):
-            yield self.err(node, 'N801', name=node.name)
+        if not name.lstrip('_')[0].isupper():
```

Comment:
I think this line will raise an `IndexError` if the class is named just `_`.

Reply:
That's a weird class name, good catch. I'll send a patch. Thanks!